addons/payment_payulatam/models/payment_transaction.py

```diff
@@ -147,6 +147,10 @@ def _process_feedback_data(self, data):
         if status == 'PENDING':
             self._set_pending(state_message=state_message)
         elif status == 'APPROVED':
+            # If the transaction is approved while beeing canceled reset the state to draft
+            # to allow set_done to work properly
+            if self.state == 'cancel':
+                self.state = 'draft'
```

Comment:
I'd rather [update the `allowed_states` in `/payment/payment_transaction.py::_set_done()` like we did for Authorize.net and Stripe](https://github.com/odoo/odoo/blob/16.0/addons/payment/models/payment_transaction.py#L682) to go from `draft` to `done` than avoiding going through `_set_done` altogether.

We'll [clean that method later in master](https://www.odoo.com/web#id=2869678&menu_id=5195&cids=1&action=333&active_id=4106&model=project.task&view_type=form).